Rakefile

```diff
@@ -14,66 +14,50 @@ task :generate_csv do
     "ocd-division/country:ug/" + parts.map { |type, value| [type, idify(value)].join(':') }.join('/')
   end
 
-  csv_url = 'https://docs.google.com/spreadsheets/d/1LetNFNq6ovg4bbq-Whze0Q06CYSpNqpRIzSUb9yVtw4/export?format=csv'
-
-  csv = CSV.parse(open(csv_url).read, headers: true, header_converters: :symbol)
-
-  ocd_ids = Set.new
-
-  csv.each do |r|
-    next unless r[:arearegion]
-    ocd_ids << {
-      id: id_for(region: r[:arearegion]),
-      name: r[:arearegion]
-    }
-
-    next unless r[:areasub_region]
-    ocd_ids << {
-      id: id_for(region: r[:arearegion], subregion: r[:areasub_region]),
-      name: r[:areasub_region]
-    }
-
-    next unless r[:areadistrict]
-    ocd_ids << {
-      id: id_for(region: r[:arearegion], subregion: r[:areasub_region], district: r[:areadistrict]),
-      name: r[:areadistrict]
-    }
-
-    next unless r[:areaconstituency]
-    ocd_ids << {
-      id: id_for(region:r[:arearegion], subregion: r[:areasub_region], district: r[:areadistrict], constituency: r[:areaconstituency]),
-      name: r[:areaconstituency]
-    }
+  def ocd_ids_from_csv(csv_data, mapping = {})
+    csv = CSV.parse(csv_data, headers: true, header_converters: :symbol)
+    ocd_ids = Set.new
+
+    csv.each do |r|
+      region = r[mapping.fetch(:region, :region)]
+      next unless region
+      ocd_ids << {
+        id: id_for(region: region),
+        name: region.strip
+      }
+
+      subregion = r[mapping.fetch(:subregion, :subregion)]
+      next unless subregion
+      ocd_ids << {
+        id: id_for(region: region, subregion: subregion),
+        name: subregion.strip
+      }
+
+      district = r[mapping.fetch(:district, :district)]
+      next unless district
+      ocd_ids << {
+        id: id_for(region: region, subregion: subregion, district: district),
+        name: district.strip
+      }
+
+      constituency = r[mapping.fetch(:constituency, :constituency)]
+      next unless constituency
+      next if ['-', '_'].include?(constituency)
```

Comment:
Hmm. It's not obvious to me what those are actually signifying. Is that just the equivalent of being blank? If so, and it's something we want to handle more generally, then we could have a more generic "is_empty?" check that looks for those as well as blank. Otherwise I think this should definitely be handled at a different level, which might be easier by splitting this up into two functions as per my wider comment.